From 126db6bda19d67d1cb2f0b65b7b514d142193a71 Mon Sep 17 00:00:00 2001
From: Ping Gao <ping.a.gao@intel.com>
Date: Fri, 29 Jan 2016 15:51:11 +0800
Subject: [PATCH 342/403] vgt/skl: redefine access policy for reg 0x20e0 and
 0x20ec

0x20e0 take as F_DOM0 and write the value from guest driver.
0x20ec take as F_RDR_MODE as it's value could be different when
workload changed.

Signed-off-by: Zhi Wang <zhi.a.wang@intel.com>
Signed-off-by: Ping Gao <ping.a.gao@intel.com>
---
 drivers/gpu/drm/i915/vgt/handlers.c |    4 ++--
 drivers/gpu/drm/i915/vgt/mmio.c     |    6 ++++++
 drivers/gpu/drm/i915/vgt/render.c   |    1 +
 3 files changed, 9 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/i915/vgt/handlers.c b/drivers/gpu/drm/i915/vgt/handlers.c
index 13a5821..37ce800 100644
--- a/drivers/gpu/drm/i915/vgt/handlers.c
+++ b/drivers/gpu/drm/i915/vgt/handlers.c
@@ -4047,8 +4047,8 @@ reg_attr_t vgt_reg_info_skl[] = {
 {0xb020, 0x80, F_RDR, 0, D_SKL, NULL, NULL},
 
 {0xd08, 4, F_VIRT, 0, D_SKL, NULL, NULL},
-{0x20e0, 4, F_PT, 0, D_SKL, NULL, NULL},
-{0x20ec, 4, F_PT, 0, D_SKL, NULL, NULL},
+{0x20e0, 4, F_DOM0, 0, D_SKL, NULL, NULL},
+{0x20ec, 4, F_RDR_MODE, 0, D_SKL, NULL, NULL},
 
 {0x4de0, 4, F_RDR, 0, D_SKL, NULL, NULL},
 {0x4de4, 4, F_RDR, 0, D_SKL, NULL, NULL},
diff --git a/drivers/gpu/drm/i915/vgt/mmio.c b/drivers/gpu/drm/i915/vgt/mmio.c
index fafe198..1c7f281 100644
--- a/drivers/gpu/drm/i915/vgt/mmio.c
+++ b/drivers/gpu/drm/i915/vgt/mmio.c
@@ -833,6 +833,12 @@ bool vgt_initial_mmio_setup (struct pgt_device *pdev)
 	pdev->initial_mmio_state[REG_INDEX(0x4dd4)] = 0x67f1427f;
 	VGT_MMIO_WRITE(pdev, 0x4dd4, 0x67f1427f);
 
+	/* TODO: need detect stepping info after pdev contain such information
+	 *  0x20e0 only enabled bit 14 after D0
+	 */
+	pdev->initial_mmio_state[REG_INDEX(0x20e0)] = 0x40004000;
+	VGT_MMIO_WRITE(pdev, 0x20e0, 0x40004000);
+
 	return true;
 }
 
diff --git a/drivers/gpu/drm/i915/vgt/render.c b/drivers/gpu/drm/i915/vgt/render.c
index 86b432c..b57d098 100644
--- a/drivers/gpu/drm/i915/vgt/render.c
+++ b/drivers/gpu/drm/i915/vgt/render.c
@@ -219,6 +219,7 @@ vgt_reg_t vgt_gen9_render_regs[] = {
 	0x2580,
 	COMMON_SLICE_CHICKEN2,
 	0x7300,
+	0x20ec,
 
 	/*this register is from NONPRIV usage*/
 	GEN8_L3SQCREG4,
-- 
1.7.10.4

